bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/dto/SensorConfig.java

```diff
@@ -23,9 +23,10 @@
  * https://dresden-elektronik.github.io/deconz-rest-doc/sensors/.
  *
  * @author David Graeff - Initial contribution
+ * @author Lukas Agethen - Add Thermostat parameters
  */
 @NonNullByDefault
-public class SensorConfig {
+public class SensorConfig extends ThermostatConfig {
```

Comment:
Why SensorConfig extends ThermostatConfig. Shouldn't this be the other way around because ThermostatConfig is the more specific class?

Reply:
I believe it is a matter of perspective. ThermostatConfig is used to send updates to deconz API and therefore it is just providing the specific parameters (otherwise the update fails).
Receiving information from the websocket service is providing all parameters, therefore I believe this is a practical solution.

Reply:
I believe this is not a matter of perspective, but it is confusing for developers who will look on this in a few weeks or month. And it will be getting even more worse if there is coming up another specific sensors later on so you will maybe get problems with multiple inheritance from specific class. For example maybe there will be a zigbee watervalve for the garden, having several config attributes and one will be `valve` you will get into trouble with your inheritance pattern. In my personal opinion inhertiance from specific to general class is an anti-pattern for this reason.

Reply:
Removed inheritance. No need to make it to scientific.